tests/Feature/Http/Controllers/ApiHeaderControllerTest.php

```diff
@@ -23,4 +23,43 @@ public function test_for_health_check()
 
         $response->assertJsonFragment($expected);
     }
+
+    public function test_header_without_value()
+    {
+        $this->withoutExceptionHandling();
+
+        $this->expectException(ApiSecretMissingException::class);
+        $this->expectExceptionMessage('Missing api header credentials');
+
+        $value = ['api_secret' => null];
+        $response = $this->json('GET','api/health-check',[], $value);
+
+        $response->assertStatus(422);
+    }
+
+    public function test_header_with_wrong_key()
+    {
+        $this->withoutExceptionHandling();
+
+        $this->expectException(ApiSecretMissingException::class);
+        $this->expectExceptionMessage('Missing api header credentials');
+
+        $value = ['api_header' => 'wrong-key'];
+        $response = $this->json('GET', 'api/health-check',[], $value);
+
+        $response->assertStatus(422);
+    }
+
+    public function test_without_header()
+    {
+        $this->withoutExceptionHandling();
+
+        $this->expectException(ApiSecretMissingException::class);
+        $this->expectExceptionMessage('Missing api header credentials');
+
+        $response = $this->json('GET', 'api/health-check',[], []);
+
+        $response->assertStatus(422);
+    }
```

Comment:
As these three tests are doing the same assertion, you can provide the input via data provider. See the examples in repo.. More [examples](https://www.w3resource.com/php/PHPUnit/writing-tests-for-phpunit-data-providers.php) here.